src/components/VendingMachine/DispenseArea/ProgressArea/ProgressItem.jsx

```diff
@@ -0,0 +1,34 @@
+import { memo } from "react";
+
+import constants from "utils/constants";
+import numberUtil from "utils/numberUtil";
+
+const { CURRENCY } = constants;
+const { seperateThousands } = numberUtil;
+
+const getProgressString = ({ type, money, product }) => {
+  const currentMoney = `${seperateThousands(money)}${CURRENCY}`;
+  let string = "";
+
+  switch (type) {
+    case "insert":
+      return `${currentMoney} 투입`;
+    case "purchase":
+      string += `${product} 구입 (${currentMoney} 사용)`;
+    // eslint-disable-next-line no-fallthrough
+    case "return":
+      string += "\n";
+      string += `${product ? "잔돈" : "투입 금액"} ${currentMoney} 반환`;
+      return string;
+    default:
+      break;
+  }
+
+  return type + money;
```

Comment:
JS에서는 변수만 보고 타입을 유추하기 힘들어서, 혹시 type과 money가 모두 string이라면 이렇게 써봐도 좋을 것 같아요!
```suggestion
  return `${type}  ${money}`;
```